src/manage_runtime_flags.c

```diff
@@ -0,0 +1,527 @@
+          && error->code == G_FILE_ERROR_NOENT)
+        {
+          g_clear_error (&error);
+          g_key_file_unref (kf);
+          return 0;
+        }
+
+      if (error)
+        {
+          g_warning ("Failed to load runtime config '%s': %s",
+                     config_path, error->message);
+          g_clear_error (&error);
+        }
+      g_key_file_unref (kf);
+      return -1;
+    }
+
+  load_feature_flag (kf, "features", "enable_agents",
+                     &out->has_agents, &out->agents);
+
+  load_feature_flag (kf, "features", "enable_container_scanning",
+                     &out->has_container_scanning, &out->container_scanning);
+
+  load_feature_flag (kf, "features", "enable_credential_store",
+                     &out->has_credential_store, &out->credential_store);
+
+  load_feature_flag (kf, "features", "enable_openvasd",
+                     &out->has_openvasd, &out->openvasd);
+
+  load_feature_flag (kf, "features", "enable_vt_metadata",
+                     &out->has_vt_metadata, &out->vt_metadata);
+
+  g_key_file_unref (kf);
+  return 0;
+}
+
+/**
+ * @brief Resolve the effective state of a single feature.
+ *
+ * Resolution order:
+ *  - If the feature is not compiled in, it is always disabled.
+ *  - If an environment variable is set and valid, use that.
+ *  - Else, if a config file value exists, use that.
+ *  - Else, default to disabled (0).
+ *
+ * @param[in,out] feature        Feature state to update.
+ * @param[in]     env_name       Environment variable name.
+ * @param[in]     conf_has_value Non-zero if configuration provided a value.
+ * @param[in]     conf_value     Value from configuration (1 or 0).
+ */
+static void
+resolve_feature (feature_state_t *feature,
+                 const char *env_name,
+                 int conf_has_value,
+                 int conf_value)
+{
+  int env_result;
+  int env_value;
+
+  if (!feature)
+    return;
+
+  if (!feature->compiled_in)
+    {
+      feature->enabled = 0;
+      return;
+    }
+
+  env_result = read_env_bool (env_name, &env_value);
+  if (env_result == 1)
+    {
+      feature->enabled = env_value;
+      return;
+    }
+
+  if (conf_has_value)
+    {
+      feature->enabled = conf_value ? 1 : 0;
+      return;
+    }
+
+  feature->enabled = 0;
+}
+
+/**
+ * @brief Append a comma-separated command list to a GString.
+ *
+ * @param[in,out] buf   Output buffer. May be NULL (then nothing is done).
+ * @param[in]     cmds  Command list to append (no leading comma).
+ */
+static void
+append_commands (GString *buf, const char *cmds)
+{
+  if (!buf)
+    return;
+
+  if (buf->len)
+    g_string_append_c (buf, ',');
+
+  g_string_append (buf, cmds);
+}
+
+/**
+ * @brief Initialize runtime feature flags from config file and environment.
+ *
+ * @param[in] config_path  Optional configuration file path, or NULL.
+ *
+ * @return Always 0 (errors are handled internally and fall back to defaults).
+ */
+int
+runtime_flags_init (const char *config_path)
```

Comment:
```suggestion
runtime_flags_init (const gchar *config_path)
```
I think we should default to gchar because we use that in more places.